#!/bin/bash
##NECESSARY JOB SPECIFICATIONS
#SBATCH --export=NONE  
#SBATCH --job-name=cactus_addGalVar
#SBATCH --time=3-00:00:00
#SBATCH --ntasks-per-node=6
#SBATCH --cpus-per-task=3
#SBATCH --mem-per-cpu=8GB
#SBATCH --output=%x.out.%j
#SBATCH --error=%x.err.%j
#SBATCH --mail-type=ALL

module load WebProxy
export SINGULARITY_CACHEDIR=$SCRATCH/.singularity

# Download cactus img file to run if not already downloaded in current directory...
if [[ ! -f cactus.img ]]
then
    singularity pull cactus.img docker://quay.io/comparative-genomics-toolkit/cactus:v2.8.2
fi

cd /scratch/user/emmarie.alexander/final_cactus_run/cactus

### cactus-update-prepare will generate a "recipe of commands" for you to run
# singularity exec cactus.img cactus-update-prepare add branch --parentGenome Anc15 --childGenome Cynocephalus_volans cactus_run11_toAddGalVar.hal cactus_run11_GalVar.txt --outDir ./steps --jobStore ./jobstore --ancestorName AncDermoptera --topBranchLength 0.084945

#### RECIPE BOOK FOR ADDING BRANCH OT PRE-EXISTING HAL GENERATED BY CACTUS-UPDATE-PREPARE ###
## generated by : /home/cactus/cactus_env/bin/cactus-update-prepare add branch --parentGenome Anc15 --childGenome Cynocephalus_volans cactus_run11_toAddGalVar.hal cactus_run11_GalVar.txt --outDir ./steps --jobStore ./jobstore --ancestorName AncDermoptera --topBranchLength 0.084945
## date : 2024-07-26 10:46:41.973879
## cactus commit : 2126683265e419d94b3efaa2f724c9f80024c3ad
## wrapping : /home/cactus/cactus_env/bin/cactus-prepare ./steps/seq_file.in --outDir ./steps --outSeqFile ./steps/seq_file.out --jobStore ./jobstore --preprocessBatchSize 1 --outHal ./steps/Anc15.hal --cactusOptions '--realTimeLogging --logInfo --retryCount 0'

## Preprocessor
singularity exec cactus.img cactus-preprocess ./jobstore/2 ./steps/seq_file.in ./steps/seq_file.out --inputNames Galeopterus_variegatus --logInfo --retryCount 0  --logFile ./steps/logs/preprocess-Galeopterus_variegatus.log

## Alignment
### Round 0
singularity exec cactus.img cactus-blast ./jobstore/3 ./steps/seq_file.out ./steps/AncDermoptera.paf --root AncDermoptera --logInfo --retryCount 0  --logFile ./steps/logs/blast-AncDermoptera.log
singularity exec cactus.img cactus-align ./jobstore/4 ./steps/seq_file.out ./steps/AncDermoptera.paf ./steps/AncDermoptera.hal --root AncDermoptera --logInfo --retryCount 0  --logFile ./steps/logs/align-AncDermoptera.log
singularity exec cactus.img hal2fasta ./steps/AncDermoptera.hal AncDermoptera --hdf5InMemory > ./steps/AncDermoptera.fa

### Round 1
singularity exec cactus.img cactus-blast ./jobstore/5 ./steps/seq_file.out ./steps/Anc15.paf --root Anc15  --logInfo --retryCount 0  --logFile ./steps/logs/blast-Anc15.log --includeRoot
singularity exec cactus.img cactus-align ./jobstore/6 ./steps/seq_file.out ./steps/Anc15.paf ./steps/Anc15.hal --root Anc15 --logInfo --retryCount 0  --logFile ./steps/logs/align-Anc15.log --includeRoot

## Alignment update
singularity exec cactus.img halAddToBranch cactus_run11_toAddGalVar.hal ./steps/AncDermoptera.hal ./steps/Anc15.hal Anc15 AncDermoptera Cynocephalus_volans Galeopterus_variegatus 0.084945 0.012688 --hdf5InMemory 

## Alignment validation
singularity exec cactus.img halValidate --genome Anc15 cactus_run11_toAddGalVar.hal --hdf5InMemory
singularity exec cactus.img halValidate --genome AncDermoptera cactus_run11_toAddGalVar.hal --hdf5InMemory
singularity exec cactus.img halValidate --genome Cynocephalus_volans cactus_run11_toAddGalVar.hal --hdf5InMemory
singularity exec cactus.img halValidate --genome Galeopterus_variegatus cactus_run11_toAddGalVar.hal --hdf5InMemory

#########################################################################################
<<SYNOPSIS
This script modifies an existing HAL alignment that was generated using Progressive Cactus by adding a new genome to the alignment via the add branch function. The add branch function works by adding the new genome to the branch connecting the parent and child nodes in the alignment. This update is a "four-hold step" and updates the given alignment at a cost of two Cactus runs. First, a new ancestor (labeled in this run as AncDermoptera) is designed.
Secondly, Cactus then infers a genome for this node. Next, Cactus aligns the recently generated AncDermoptera with Cynocephalus volans. This step produces a Anc15.hal, containing the updated alignment information for this node. Lastly, the halAddTOBranch tool comes into play to update the alignment blocks of Anc15 in the initial alignment HAL file. This procedure should not change the Anc15 or Anc24 genomes. The genomes represented by these should remain the same.
SYNOPSIS

<<CITATIONS
	  - TAMU HPRC: https://hprc.tamu.edu/research/citations.html
	  - Progressive Cactus: https://github.com/ComparativeGenomicsToolkit/cactus
		- Updating Cactus alignments: https://github.com/ComparativeGenomicsToolkit/cactus/blob/master/doc/updating-alignments.md
		- cactus-update-prepare: https://github.com/ComparativeGenomicsToolkit/cactus/blob/master/doc/cactus-update-prepare.md
CITATIONS
